index.html

```diff
@@ -1,138 +1,176 @@
+	</style>
 </head>
 <body>
 
 <h1>GPG Online</h1>
 
 <div class="message">
-  <h2>Message to encrypt</h2>
-  <textarea rows="5"></textarea>
+	<h2>Message to encrypt</h2>
+	<textarea rows="5"></textarea>
 </div>
 
 <div id="public-keys">
-  <div class="public-key">
-    <h2>Public key</h2>
-    <p class="hint">Only the person who has a private key will be able to decrypt the message.</p>
-    <textarea rows="5"></textarea>
-  </div>
+	<div class="public-key">
+		<h2>Public key</h2>
+		<p class="hint">Only the person who has a private key will be able to decrypt the message.</p>
+		<textarea rows="5"></textarea>
+	</div>
 </div>
 
 <div id="private-key">
-  <h2>Private Key</h2>
-  <textarea rows="5"></textarea>
-  <h2>Passphrase</h2>
-  <textarea rows="2"></textarea>
+	<h2>Private Key</h2>
+	<textarea rows="5"></textarea>
+	<h2>Passphrase</h2>
+	<textarea rows="2"></textarea>
 </div>
 
-<pre id="output"></pre>
+<div>
+	<input id="uploader" type="file" />
+</div>
+<div id="output"></div>
 <button id="encrypt">Encrypt</button>
 <button id="add-public-key">Add Public Key</button>
 <button id="add-private-key">Add Private Key</button>
-
 <script>
 const query = query => document.querySelector(query);
 const queryAll = query => Array.prototype.slice.apply(document.querySelectorAll(query));
 const publicKeyTemplate = query('#public-keys').innerHTML;
 
+
+query("#uploader").addEventListener('change', function() {
+
+	var reader = new FileReader();
+	reader.onload = async function() {
+
+		var arrayBuffer = this.result;
+		var fileData = new Uint8Array(arrayBuffer);
+
+		var pubKeyData = queryAll('.public-key textarea').map(el => getKey(el, 'public'))
+		pubKey = (await openpgp.key.readArmored(pubKeyData[0])).keys[0];
+		const options = {
+			message: openpgp.message.fromBinary(fileData),
+			publicKeys: pubKey,
+			privateKey: "",
+			armor: true,
+		};
+
+		return openpgp.encrypt(options).then(results => {
+			link = document.createElement('a');
+			link.setAttribute('href', window.URL.createObjectURL(
+			new Blob([results.data]), {type: "application/pgp-encrypted"})
+			);
+			link.setAttribute('download', "encrypted.file.gpg");
+			link.innerHTML="encrypt.png"
+			query('#output').appendChild(link);
+		});
+
+
+	}
+	reader.readAsArrayBuffer(this.files[0]);
+
+}, false);
+
+
 query('#add-public-key').addEventListener('click', () => {
-  const element = document.createElement('div');
-  element.innerHTML = publicKeyTemplate;
-  query('#public-keys').appendChild(element);
+	const element = document.createElement('div');
+	element.innerHTML = publicKeyTemplate;
+	query('#public-keys').appendChild(element);
 });
 
 query('#add-private-key').addEventListener('click', () => {
-  query('#private-key').className = 'visible';
+	query('#private-key').className = 'visible';
 });
 
 function getKey(textarea, type) {
-  const { value } = textarea;
-  if (!value) {
-    throw new Error(`${type} key missing`);
-  }
-  return openpgp.key.readArmored(value).keys[0];
```

Comment:
Curious why was this removed? `getKey` now doesn't really do much.

Reply:
Hi Alex, 
as I warned before... I am a JS noob. I had troubles with the new async API.
So I removed the key reading to it's own block. I removed the Error throwing too just out of frustration I guess.
I now pushed a new working implementation. You can find it in my Fork. 
This implementation can also encrypt a file. But I am still trying to figure out how to rename the file properly.

Reply:
my current code works if I debug it. So I think I am doing something wrong with Promises. But I don't grok anything with async await in JS yet. So I let it be for the while...